### PR TITLE
Maya2015

### PR DESCRIPTION
**Pull request by rhaleblian**
This gets us a building and running plugin against the niftools fork.
Solution design stripped down to be able to practically maintain many different Mayas.
Tested with 2015-x64, 2013-win32, 2013-x64, by importing, editing and exporting Skyrim childhead.
I have no idea what the full feature set is, but UVs were preserved so yay.

Git-flow would dictate that you pull this into niftools:maya2015 and not niftools:master, yes?

### CODE REVIEW

**Comment by neomonkeus** [comment]
Yeah you are correct, at the very least this should be pull requested against develop, not master.

We also need to discuss generally what the "main track" versions should be supported, i.e either develop stays pre-2012, where there should be a new branch for maya2015 support or if pre-2012 should be branched off and maya2015 is the main supported branch of development.

Also what range of compatability is this maya2015 branch. Does it cover 2013-2015 versions?

Edit : Re-read the full description, so this is 2013-2015 compatible so. In that case we can make a branch for maya2015. 

Not sure if you are able to change the branch you have the pull request against. If not I can branch off develop and swap the names.

**Comment by rhaleblian** [comment]
As I try to figure out how to point to the right base branch, I might close this request.

2012 win32 and x64 should also work.

**Comment by neomonkeus** [comment]
If is compatible with Maya 2012 then we could probably merge it develop. Not sure how backwards compatible the API is beyond that, don't have any experience with Maya.

Perhaps we should get some builds out there first and see what happens?

**Comment by rhaleblian** [comment]
Yes. Is there a location where I can upload installers?

**Comment by neomonkeus** [comment]
If you have dropbox I can share a folder with you, if not we can do it another way. Probably the best thing to do is pm me on the forums.

**Comment by rhaleblian** [comment]
Found a silly bug that breaks exporting, I need to fix it, as a new pull request.

File: src/Exporters/NifMaterialExporterSkyrim.cpp [hunk 1]
```diff
@@ -25,9 +25,12 @@ void NifMaterialExporterSkyrim::ExportMaterials() {
 			string texture = "";
 			string normal_map = "";
 
-			unsigned int shader_type = 0;
-			unsigned int shader_flags1 = 0;
-			unsigned int shader_flags2 = 0;
+			//BSLightingShaderPropertyShaderType shader_type = DEFAULT;
+			//SkyrimShaderPropertyFlags1 shader_flags1 = SLSF1_SPECULAR;  // plateofshrimp: in lieu of missing NONE value
```
Comment (neomonkeus) [inline src/Exporters/NifMaterialExporterSkyrim.cpp:29]: No need to include your name here, git auto-tags delta object, this is visible using git blame.

File: src/Exporters/NifMaterialExporterSkyrim.cpp [hunk 2]
```diff
@@ -122,39 +125,39 @@ void NifMaterialExporterSkyrim::ExportMaterials() {
...
-			shader_property->setShaderFlags1(shader_flags1);
-			shader_property->setShaderFlags2(shader_flags2);
-			shader_property->setUnknownInt7(3);
-			shader_property->setAlpha(alpha);
+			shader_property->SetSkyrimShaderType((BSLightingShaderPropertyShaderType)shader_type);
+			shader_property->SetTextureSet(shader_textures);
+			shader_property->SetGlossiness(glossiness);
+			shader_property->SetEmissiveColor(emissive_color);
+			shader_property->SetSpecularColor(specular_color);
+			//shader_property->SetTextureTranslation1(texture_translation1); FIXME
```
Comment (neomonkeus) [inline src/Exporters/NifMaterialExporterSkyrim.cpp:154]: Can you open issue(s) for this, if it was just broken though general NifLIb updates where the fix is indeed required on Maya's end. If its the converse where the issue should be fixed in NifLib and tracking issue be opened on Maya to resolve once fixed there.

File: src/Importers/NifMaterialImporter.cpp [hunk 3]
```diff
@@ -275,6 +275,7 @@ void NifMaterialImporter::ImportMaterialsAndTextures( NiAVObjectRef & root )
 				}
 
 				//Create 2D Texture Placement
+				//MFnDependencyNode *texture_placement = new MFnDependencyNode();
```
Comment (neomonkeus) [inline src/Importers/NifMaterialImporter.cpp:278]: Is this just clean up required? Should remove commented code if not needed

**Pull request closed without merging**